src/exitFinalizer/exitFinalizer.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright (c) 2019-present, LeapDAO (leapdao.org)
+ *
+ * This source code is licensed under the Mozilla Public License Version 2.0
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+
+/* eslint-disable no-console */
+
+import { Exit, helpers } from 'leap-core';
+import { lessThan, divide, multiply, bi } from 'jsbi-utils';
+import { Errors } from 'leap-lambda-boilerplate';
+import getToken from '../common/getToken';
+
+const { getProof } = helpers;
+
+const { BadRequest, ServerError } = Errors;
+
+class ExitFinalizer {
+
+  constructor(rate, senderAddr, exitHandler, operator, root, plasma, db, marketConfig) {
+    this.rate = rate;
+    this.senderAddr = senderAddr;
+    this.exitHandler = exitHandler;
+    this.operator = operator;
+    this.db = db;
+    this.root = root;
+    this.plasma = plasma;
+    this.marketConfig = marketConfig;
+  }
+
+  async sellExits() {
+    const latestBlock = await this.plasma.getBlockNumber();
+    const latestPeriod = Math.floor(latestBlock / 32);
+    console.log(`Latest block: ${latestBlock}. Latest period: ${latestPeriod}`);
+
+    const result = {};
+
+    await Promise.all(this.marketConfig.map(async (market) => {
+      const exits = await this.db.getProvableExits(market.color, latestBlock);
+      console.log('Sold exits to process:', exits.length);
+
+      result[market.color] = { total: exits.length, done: 0 };
+
+      for (let i = 0; i < exits.length; i += 1) {
+        const exit = exits[i];
+        await this.sellExit(exit).then((rsp) => { // eslint-disable-line no-await-in-loop
+          console.log('Processed sold exit:', exit.utxoId, rsp);
+          result[market.color].done += 1;
+          return this.db.setAsFinalized(exit.utxoId, rsp.hash);
+        }).catch(e => console.error(e));
+      }
```

Comment:
Rare case when `no-await-in-loop` rule really helps. Now we waiting for each exit, but maybe would be a good idea to use Promise.all here as well?

Like

```es6
await Promise.all(exits.map(exit => {
  return this.sellExit(exit).then((rsp) => { // eslint-disable-line no-await-in-loop
    console.log('Processed sold exit:', exit.utxoId, rsp);
    result[market.color].done += 1;
    return this.db.setAsFinalized(exit.utxoId, rsp.hash);
  }).catch(e => console.error(e))
}));
```

Reply:
I had the code like this initially, but reworked to sync, because a proper nonce management is required (sellExits sends a transaction) and I haven't figured out the easy way (lame)

Reply:
I see. That's good then 👍